page-tracking/src/page_tracker.rs

```diff
@@ -733,6 +733,39 @@ impl PageTracker {
         true
     }
 
+    /// Returns true if `addr` is a page owned by `owner` and it's in a "Blocked" state with
+    /// type `mem_type`, or if `addr` is a page not owned by `owner` and it's in a "SharedBlocked"
+    /// state with type `mem_type`. Additionally, the TLB version of the page must be older than
+    /// the version provided by the caller.
+    pub fn is_blocked_page_with_older_tlb(
```

Comment:
this is basically the same check as `is_removable_page`. can we combine them into a single function?